resident/resident-service/src/main/java/io/mosip/resident/service/impl/IdentityServiceImpl.java

```diff
@@ -397,5 +397,45 @@ String getIndividualIdForAid(String aid)
 			return individualId;
 	}
 
+	public String getClaimFromIdToken() throws ApisResourceAccessException {
+		AuthUserDetails authUserDetails= getAuthUserDetails();
+		String claim = this.environment.getProperty(ResidentConstants.IDA_TOKEN_CLAIM_NAME);
+		String claimValue = "";
+		String idToken = authUserDetails.getIdToken();
+		if(idToken!=null){
+			String payLoad = decodeTokenParts(idToken);
+			System.out.println(payLoad);
+			Map<String, Object> payLoadMap=convertStringToMap(payLoad);
+			if(claim!=null){
+				claimValue = (String) payLoadMap.get(claim);
+			}
+		}
+		return  claimValue;
+	}
+
+	private Map<String, Object> convertStringToMap(String payLoad) {
```

Comment:
Use ObjectMapper to convert it to a map.

Reply:
Tried ObjectMapper `Map<String, Object> payLoadMap= objectMapper.convertValue(payLoad, Map.class);`
like this but giving error `Cannot construct instance of `java.util.LinkedHashMap` (although at least one Creator exists): no String-argument constructor`.
So implemented this method covertStringToMap()

Reply:
Done.